tests/unit/lib/FluxibleContext.js

```diff
@@ -148,13 +149,84 @@ describe('FluxibleContext', function () {
+                                cb();
+                            });
+                        }
+                    ], function (err) {
+                        callback(err)
+                    });
+                };
+                actionOne.displayName = 'One';
+                var actionTwo = function (context, payload, callback) {
+                    actionCalls.push({
+                        context: context,
+                        payload: payload
+                    });
+                    callback();
+                };
+                actionTwo.displayName = 'Two';
+                var actionThree = function (context, payload, callback) {
+                    actionCalls.push({
+                        context: context,
+                        payload: payload
+                    });
+                    callback();
+                };
+                actionThree.displayName = 'Three';
+                var payload = {};
+                var callback = function () {
+                    expect(actionCalls.length).to.equal(4);
+                    expect(actionCalls[0].context).to.contain.keys(Object.keys(actionContext));
+                    expect(actionCalls[0].context).to.contain.keys(['id','stack']);
+                    expect(actionCalls[0].context.id).to.equal('THIS_IS_UNIQUE');
+                    expect(actionCalls[0].context.stack.join('.')).to.equal('Action.One');
+                    expect(actionCalls[0].payload).to.equal(payload);
+                    expect(actionCalls[1].context).to.contain.keys(Object.keys(actionContext));
+                    expect(actionCalls[1].context).to.contain.keys(['id','stack']);
+                    expect(actionCalls[1].context.id).to.equal('THIS_IS_UNIQUE');
+                    expect(actionCalls[1].context.stack.join('.')).to.equal('Action.One.Two');
+                    expect(actionCalls[1].payload).to.equal(payload);
+                    expect(actionCalls[2].context).to.contain.keys(Object.keys(actionContext));
+                    expect(actionCalls[2].context).to.contain.keys(['id','stack']);
+                    expect(actionCalls[2].context.id).to.equal('THIS_IS_UNIQUE');
+                    expect(actionCalls[2].context.stack.join('.')).to.equal('Action.One.Two');
+                    expect(actionCalls[2].payload).to.equal(payload);
+                    expect(actionCalls[3].context).to.contain.keys(Object.keys(actionContext));
+                    expect(actionCalls[3].context).to.contain.keys(['id','stack']);
+                    expect(actionCalls[3].context.id).to.equal('THIS_IS_UNIQUE');
+                    expect(actionCalls[3].context.stack.join('.')).to.equal('Action.One.Three');
+                    expect(actionCalls[3].payload).to.equal(payload);
+                    done();
+                };
+                actionContext = context.getActionContext({
+                    id: 'THIS_IS_UNIQUE'
```

Comment:
For testing since we won't allow it to be passed in, you can mock out the generatUUID method.